libs/payments/paypal/src/lib/paypal.manager.ts

```diff
@@ -47,11 +47,20 @@ export class PayPalManager {
     };
   }
 
+  /**
+   * Retrieves the customer’s current paypal billing agreement ID from the
+   * auth database via the Paypal repository
+   */
+  async getCustomerBillingAgreementId(customer: Stripe.Customer) {
+    const paypalCustomer =
+      await this.paypalCustomerManager.fetchPaypalCustomersByUid(
+        customer.metadata.userid
+      );
+    return paypalCustomer[0].billingAgreementId;
```

Comment:
Couple of things here:

1. I think we should throw an error if more than one paypalCustomer is found (it's an unexpected scenario).
2. This code doesn't currently cover when no paypalCustomer records are returned (it throws, since it cannot read property `billingAgreementId` of undefined.
  - A note here is that `[0]` is actually not _technically_ fully type-safe. The result type of `paypalCustomer[0]` is `PaypalCustomer` in this instance, but it really should be `PaypalCustomer | undefined` (since array access could result in no value!). This is an acknowledged Typescript _thing_, and they decided to leave it like this since it would break too much legacy code (there's a Github issue about this if you're curious). Instead, they recommend using `.at(0)` (or another index) for array access for fully type-safe array access.